### DISCUSSION TITLE
Any key to make a "powerOnState" for a zigbee switch?

### DISCUSSION THREAD

**Discussion by ortegafernando** [arendst/Tasmota#15987] [Q&A] [answered]
Hi, has anybody a key or idea to implement something like a "poweOnState" for a zigbee switch?

Thanks.

**Comment by supperka** [comment]
save status in node red, than make automation.
add inject node trigger after NR starts than control if state was on, turn on or off
but turn on save variables on disk.

**Comment by s-hadinger** [comment, marked as answer]
There is an optional attribute in Zigbee for poweronstate. Let me find it shortly

**Reply by s-hadinger** [reply]
There is an optional attribute `StartUpOnOff` to specify the value at power on. It's optional in the zigbee spec, so your device may not support it.

`zbsend {"device":"<your_device>","write":{"StartUpOnOff":0}}`  (off)

or

`zbsend {"device":"<your_device>","write":{"StartUpOnOff":1}}`  (on)

**Reply by ortegafernando** [reply]
Hi, thanks a lot for your proposal but I can't get managed.

```
16:08:02.235 CMD: zbsend {"device":"Switch1","write":{"StartUpOnOff":0}}
16:08:02.959 MQT: ch/domo/ZBBRIDGE/stat/RESULT = {"ZbSend":"Done"}
16:08:03.702 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbResponse":{"Device":"0xF784","Name":"Switch1","Command":"0006_04","Status":134,"StatusMessage":"UNSUPPORTED_ATTRIBUTE","Endpoint":1,"LinkQuality":29}}
16:08:24.242 CMD: zbsend {"device":"Switch1","write":{"StartUpOnOff":1}}
16:08:24.961 MQT: ch/domo/ZBBRIDGE/stat/RESULT = {"ZbSend":"Done"}
16:08:25.800 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbResponse":{"Device":"0xF784","Name":"Switch1","Command":"0006_04","Status":134,"StatusMessage":"UNSUPPORTED_ATTRIBUTE","Endpoint":1,"LinkQuality":24}}
```

I get "UNSUPPORTED_ATTRIBUTE". I don't know if this message is from the device itselft or from tasmota firmware. I can't see StartUpOnOff command in https://tasmota.github.io/docs/Zigbee/#sending-device-commands.

Could be it has not been implemented in tasmota? (I have searched source code and I can find StartUpOnOff, but I don't know exactly if it is completly implemented or not. Also, as I haven't read about it in tasmota docs....)

Thanks a lot.

NOTE: My device is an sonoff zigbee switch. (ZB MINI)

**Reply by sfromis** [reply]
Looks like a message from the device itself. StartUpOnOff is not a Tasmota feature, it is an attribute of the device. As s-hadinger wrote, not supported by all devices.

If the device sends an incoming message upon power up (?), you could make an automation (like rules) to react to that. If it does not, how would Tasmota know?

**Reply by ortegafernando** [reply]
Ok, thanks a lot . I will need to implement in nodered, thanks

**Reply by s-hadinger** [reply]
Yes, the message comes from the device itself.

It's unclear from https://github.com/dresden-elektronik/deconz-rest-plugin/issues/4016, because the device seems to announce supporting this attribute. However Z2M also fails at setting this attribute.

I would like to try adding a manufacturer id since it's sometimes required for specific attributes, but I can't find any ID for Sonoff/Itead.

**Comment by s-hadinger** [comment]
One last try, I found here https://github.com/stas-demydiuk/domoticz-zigbee2mqtt-plugin/issues/517 that manufacturerId is 0x1002 (I have no clue how they found this value).

Can you please try:

```
zbsend {"device":"Switch1","manuf":"0x1002","write":{"StartUpOnOff":0}}
```

**Reply by ortegafernando** [reply]
Thanks a lot for the try, but no luck.

```
21:15:18.795 CMD: zbsend {"device":"Switch1","manuf":"0x1002","write":{"StartUpOnOff":0}}
21:15:19.406 MQT: ch/domo/ZBBRIDGE/stat/RESULT = {"ZbSend":"Done"}
21:15:20.122 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbResponse":{"Device":"0xF784","Name":"Switch1","Command":"0006_04","Status":134,"StatusMessage":"UNSUPPORTED_ATTRIBUTE","Endpoint":1,"LinkQuality":24}}
21:15:33.731 CMD: zbsend {"device":"Switch1","manuf":"0x1002","write":{"StartUpOnOff":1}}
21:15:34.342 MQT: ch/domo/ZBBRIDGE/stat/RESULT = {"ZbSend":"Done"}
21:15:35.072 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbResponse":{"Device":"0xF784","Name":"Switch1","Command":"0006_04","Status":134,"StatusMessage":"UNSUPPORTED_ATTRIBUTE","Endpoint":1,"LinkQuality":24}}
```

Here it seems also that it can't be done with ZBMINI: https://github.com/Koenkk/zigbee2mqtt/issues/5901

**Reply by s-hadinger** [reply]
Too bad.

**Reply by ortegafernando** [reply]
Hi, about making logic in nodered... I know about INFO1, INFO2, INFO3 mqtt payloads, but thet are "from" zigbee bridge.

How could I detect the "first" message of a zigbee node after start up? 

Something about ZbParent may be? I can't find any info.

Thanks

**Reply by barbudor** [reply]
You may use a flow context variable
- on receiving the info message erase it
- on receiving message from a given device check if the variable has a flag for this device and set the flag anyhow. If the flag was not there, it was the 1st message

**Reply by s-hadinger** [reply]
The message indicating that Zigbee started is

```
13:12:38.046 RSL: RESULT = {"ZbState":{"Status":0,"Message":"Started"}}
```

So you can trigger on ZbState#Status == 0

**Reply by sfromis** [reply]
What exactly is the situation you're asking about? The Zigbee device being powered up, the bridge being booted, the bridge connecting, or what? Answers seems to indicate that this is not clear, and my answers previously was based on a "power on state" for the Zigbee device, irrespective of what's happening on the bridge. At least that fits what you could get if the Zigbee device supported such a function. And that's why I pointed out the importance of any message arriving from the Zigbee device after power up, as this would be great for automation. Are you seeing any messages then?

**Reply by ortegafernando** [reply]
> You may use a flow context variable
> 
> * on receiving the info message erase it
> * on receiving message from a given device check if the variable has a flag for this device and set the flag anyhow. If the flag was not there, it was the 1st message

Ok, great option!

**Reply by ortegafernando** [reply]
> The message indicating that Zigbee started is
> 
> ```
> 13:12:38.046 RSL: RESULT = {"ZbState":{"Status":0,"Message":"Started"}}
> ```
> 
> So you can trigger on ZbState#Status == 0

This solve when the "bridge" has start, but I also need about the zigbee "sensor" itself. barbudor option is good

**Reply by ortegafernando** [reply]
> What exactly is the situation you're asking about? The Zigbee device being powered up, the bridge being booted, the bridge connecting, or what? Answers seems to indicate that this is not clear, and my answers previously was based on a "power on state" for the Zigbee device, irrespective of what's happening on the bridge. At least that fits what you could get if the Zigbee device supported such a function. And that's why I pointed out the importance of any message arriving from the Zigbee device after power up, as this would be great for automation. Are you seeing any messages then?

I agree you. My question is not clear. Sorry

I want to apply my "own power on state" in a specific device, an SONOFF ZBMINI.

Ideal solution will be to detect when the SONOFF ZBMINI is powerup.
Not ideal, but good solution, will be to detect the zigbee bridge powerup (barbudor option):

About "Ideal solution will be to detect when the SONOFF ZBMINI is powerup". may be the only way is to "log" this:

```
zbinfo <device>
```
And check "LastSeen" from the json answer

```
{"ZbInfo":{"0xF784":{"Device":"0xF784","Name":"Switch1","IEEEAddr":"0xB4E3F9FFFEF9692D","ModelId":"01MINIZB","Manufacturer":"SONOFF","Endpoints":[1],"Config":["O01"],"Power":1,"Reachable":true,"LastSeen":225,"LastSeenEpoch":1660170818,"LinkQuality":31}}}
```

If 'LastSeen' is > than X then may be the next time you get sensor data from that device ... it just start up again. (or LastSeenEpoch)
(LastSeen | Number of seconds since the last message was received
LastSeenEpoch | Timestamp when the last message was received)

I mean:
1. call every X minutes "zbinfo switch1"
2. read LastSeen value
3. if LastSeen > Y seconds => device is powered off or has lost zigbee signal. Go to step 5
4. if lastSeen < Y seconds, no problem, go to step 1
5. Wait to next ZbReceived sensor data from that device and "apply your own POWER ON START UP"

What do you think, guys?

**Reply by sfromis** [reply]
Well, did you check if anything arrives when powering the ZbMini up? If this happens, reacting to that is a much better way of doing it.

If nothing arrives, and you do not need a "power on state" after all, but instead a delayed detection, your approach may work, but first you do have to check whether you actually get any *recent* "last seen" quickly after powering the ZbMini up? Only after knowing what you have to work with can it make sense to construct an algorithm.

**Reply by barbudor** [reply]
One possible issue is if you are recovering from a global power outage, zigbee devices will power up probably almost immediately quickly followed by Tasmota devices.
The wifi network may take up to 1 minute and the node red host may be 2 or more.

**Reply by ortegafernando** [reply]
> Well, did you check if anything arrives when powering the ZbMini up? If this happens, reacting to that is a much better way of doing it.
> 
> If nothing arrives, and you do not need a "power on state" after all, but instead a delayed detection, your approach may work, but first you do have to check whether you actually get any _recent_ "last seen" quickly after powering the ZbMini up? Only after knowing what you have to work with can it make sense to construct an algorithm.

Hi, I have update first to 12.0.2 tasmota firmware. I still have the 6.7.8 zigbee firmware.

I have done this test:
1. power down and up the ZBMINI
2. power down and up a water leak sensor
3. power down and up the ZBMINI

I get this in console (sorry about first boot and info lines lost)

I can get this message in console everytime I power up the ZBMINI. 
```
ZIG: {"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}
```
but after 20 seconds (aprox) since booting it up.

The water leak sensor doesn't show the same behaviour.

My objective is to have a start up condition for heating, so 20 seconds is not a problem, even 5 minutes as Barbudor suggested.

I could log every mqtt payload, but I need to recieve in nodered ZbEZSPReceived.

I have tried this rule (and rule1 ON, of course)
```
ON ZbEZSPReceived#Data DO var1 %value% ON ENDON
```
But it doesn't fire.

Thanks

```
08:12:12.128 MQT: ch/domo/ZBBRIDGE/tele/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":1983}}
08:12:14.886 ZIG: Resetting EZSP device
08:12:16.766 MQT: ch/domo/ZBBRIDGE/tele/STATE = {"Time":"2022-08-11T08:12:16","Uptime":"0T00:00:18","UptimeSec":18,"Vcc":3.512,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":25,"MqttCount":1,"Wifi":{"AP":1,"SSId":"XXXX","BSSId":"4C:4E:03:00:22:AC","Channel":1,"Mode":"11n","RSSI":100,"Signal":-23,"LinkCount":1,"Downtime":"0T00:00:03"}}
08:12:17.444 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbState":{"Status":1,"Message":"EFR32 EZSP booted","RestartReason":"Power-on","Code":2}}
08:12:18.182 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbState":{"Status":55,"Version":"6.7.8.0","Protocol":8,"Stack":2}}
08:12:18.962 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbState":{"Status":3,"Message":"Configured, starting coordinator"}}
08:12:21.063 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbState":{"Status":56,"IEEEAddr":"0x804B50FFFE058A41","ShortAddr":"0x0000","DeviceType":1}}
08:12:21.124 ZIG: Subscribe to group 0 'ZbListen0 0'
08:12:23.683 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbState":{"Status":0,"Message":"Started"}}
08:12:23.684 ZIG: Zigbee started
08:12:23.714 ZIG: Zigbee device information found in EEPROM (5 devices - 196 bytes)
08:12:23.754 ZIG: Zigbee device data in EEPROM (113 bytes)
08:12:36.909 MQT: ch/domo/ZBBRIDGE/tele/RESULT = {"ZbParent":{"Device":"0x0000","Children":4,"ChildInfo":["0x5C0272FFFEBC0546","0x847127FFFE18FB4B","0x00124B0022FEB6BB","0x00124B0022E96ED9"]}}
08:14:39.570 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0x8AAD":{"Device":"0x8AAD","Name":"PIR","0500<00":"010000010000","ZoneStatusChange":1,"ZoneStatusChangeZone":1,"Occupancy":1,"Endpoint":1,"LinkQuality":113}}}
08:16:21.063 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0x8AAD":{"Device":"0x8AAD","Name":"PIR","0500<00":"000000010000","ZoneStatusChange":0,"ZoneStatusChangeZone":1,"Occupancy":0,"Endpoint":1,"LinkQuality":113}}}
08:16:27.702 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":11}}}
08:16:47.282 ZIG: {"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}
08:17:17.787 MQT: ch/domo/ZBBRIDGE/tele/STATE = {"Time":"2022-08-11T08:17:17","Uptime":"0T00:05:19","UptimeSec":319,"Vcc":3.513,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"XXXX","BSSId":"4C:4E:03:00:22:AC","Channel":1,"Mode":"11n","RSSI":100,"Signal":-25,"LinkCount":1,"Downtime":"0T00:00:03"}}
08:20:23.137 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":14}}}
08:20:43.719 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0x3218":{"Device":"0x3218","Name":"Temp","Temperature":25.1,"Endpoint":1,"LinkQuality":94}}}
08:20:44.846 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0x3218":{"Device":"0x3218","Name":"Temp","Humidity":66.7,"Endpoint":1,"LinkQuality":94}}}
08:22:17.927 MQT: ch/domo/ZBBRIDGE/tele/STATE = {"Time":"2022-08-11T08:22:17","Uptime":"0T00:10:19","UptimeSec":619,"Vcc":3.512,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"XXXX","BSSId":"4C:4E:03:00:22:AC","Channel":1,"Mode":"11n","RSSI":100,"Signal":-24,"LinkCount":1,"Downtime":"0T00:00:03"}}
08:23:07.229 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xDEA3":{"Device":"0xDEA3","Name":"Temp2","Humidity":71.42,"Endpoint":1,"LinkQuality":19}}}
08:24:22.422 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":16}}}
08:27:17.783 MQT: ch/domo/ZBBRIDGE/tele/STATE = {"Time":"2022-08-11T08:27:17","Uptime":"0T00:15:19","UptimeSec":919,"Vcc":3.512,"Heap":20,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Wifi":{"AP":1,"SSId":"XXXX","BSSId":"XXXX","Channel":1,"Mode":"11n","RSSI":100,"Signal":-24,"LinkCount":1,"Downtime":"0T00:00:03"}}
08:28:22.062 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":14}}}
08:28:25.524 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0x8AAD":{"Device":"0x8AAD","Name":"PIR","0500<00":"010000010000","ZoneStatusChange":1,"ZoneStatusChangeZone":1,"Occupancy":1,"Endpoint":1,"LinkQuality":113}}}
08:30:03.234 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":32}}}
08:30:04.320 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":27}}}
08:30:26.633 ZIG: {"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}
08:30:32.315 MQT: ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0x8AAD":{"Device":"0x8AAD","Name":"PIR","0500<00":"000000010000","ZoneStatusChange":0,"ZoneStatusChangeZone":1,"Occupancy":0,"Endpoint":1,"LinkQuality":107}}}
```

**Reply by barbudor** [reply]
What I wanted to point is that if node-red boots minutes after zbmini and zbbridge it will not see those Zb boot messages

**Reply by sfromis** [reply]
> I can get this message in console everytime I power up the ZBMINI.

Was that the **only** activity related to powering up again?

Power cycling another relay-type Zigbee device (the Aqara T1), I get a bunch of messages. including a regular `.../SENSOR` indicating that the device is present again.

Note that rules can only trigger on the payloads for MQTT, those that are tagged with `MQT:` in the console. And those are also the only ones available to MQTT subscribers, like Node-Red. One potential trick for the backend could be to activate `MqttLog` and have all the console-eligible messages at the selected log level replicated to another MQTT topic, but doing it that way is fragile as other messages are not meant as a stable API.

Assuming that your ZbMini is `0xF784` (I can strongly recommend using ZbName to assign names), in both instances of `{"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}` you also have a SENSOR payload about 20 seconds previously. But you did not indicate the timing of the log vs what you did to the power situation.
`ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":27}}}`

That payload would be excellent for seeing that the device is present and powered off. Good input for powering up again.

Of course, there still is the issue pointed out by barbudor, about messages lost when a whole-house outage would likely lead to the Zigbee device booting first, with no one listening, as the bridge would likely complete booting a few seconds later, and the backend server with Node-Red even later. 

If your objective is to just keep it on all the time, you could also have the booting of the bridge and/or Node-Red issue a power on command for good measure. If you also have a desired state, you could somewhere store the desired state to be enforced upon booting.

**Reply by ortegafernando** [reply]
> What I wanted to point is that if node-red boots minutes after zbmini and zbbridge it will not see those Zb boot messages

Yes, you are right also. Usually yes. In may case I have nodered in a external hosting, but it will occur also as the router will need more time to boot than the Zb devices (sensors and bridge).

**Reply by ortegafernando** [reply]
> > I can get this message in console everytime I power up the ZBMINI.
> 
> Was that the **only** activity related to powering up again?
> 
> Power cycling another relay-type Zigbee device (the Aqara T1), I get a bunch of messages. including a regular `.../SENSOR` indicating that the device is present again.
> 
> Note that rules can only trigger on the payloads for MQTT, those that are tagged with `MQT:` in the console. And those are also the only ones available to MQTT subscribers, like Node-Red. One potential trick for the backend could be to activate `MqttLog` and have all the console-eligible messages at the selected log level replicated to another MQTT topic, but doing it that way is fragile as other messages are not meant as a stable API.
> 
> Assuming that your ZbMini is `0xF784` (I can strongly recommend using ZbName to assign names), in both instances of `{"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}` you also have a SENSOR payload about 20 seconds previously. But you did not indicate the timing of the log vs what you did to the power situation. `ch/domo/ZBBRIDGE/tele/SENSOR = {"ZbReceived":{"0xF784":{"Device":"0xF784","Name":"Switch1","Power":0,"Endpoint":1,"LinkQuality":27}}}`
> 
> That payload would be excellent for seeing that the device is present and powered off. Good input for powering up again.
> 
> Of course, there still is the issue pointed out by barbudor, about messages lost when a whole-house outage would likely lead to the Zigbee device booting first, with no one listening, as the bridge would likely complete booting a few seconds later, and the backend server with Node-Red even later.
> 
> If your objective is to just keep it on all the time, you could also have the booting of the bridge and/or Node-Red issue a power on command for good measure. If you also have a desired state, you could somewhere store the desired state to be enforced upon booting.

Hi, only to put it more clear:

I recieve this "`{"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}`" after about 20 seconds of the very first SENSOR message from the ZBMINI when it boots up.

I mean: it boots up, then I recieve a SENSOR message and after 20 seconds (more or less) I get that ZbEZSPReceived.

As you both say, it will be dificult to detect always the boot up event this way.

But I have an advantage, my nodered instance is in an external web hosting, so probably, it will be ontime always (almost).

So may be this is my solution:

1. nodered is in web hosting, outside "my home". (even my router has a little battery to survive to short blackouts)
2. if anything happens to router or zigbee bridge I will recieve again INFO1, INFO2 and INFO3 messages from zigbee bridge, so I can use them to "resend" the poweronstate (it is not 100% sure, because ZBMINI could be ok, but we can't ensure all user cases)
3. if anything happens to ZBMINI I could get perfectly the SENSOR message after it boots up and 20 seconds after I will get the "{"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}". For example my wifi switch down and up the heating main breaker.

May be it is not "100% pure", but it is a good approach

But I need to solve about receiving {"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}". Is "mqttlog" the only option available? will be difficult to implement a new MQT message with ZbEZSPReceived?

NOTE: @sfromis I have configured a ZbName: "switch1"

**Reply by sfromis** [reply]
Why are you even looking so much for that `{"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}"` message? I'd just ignore it, unless you see some very specific good argument for even worrying about that one.

If you get the `...SENSOR` payload, including `"Power":0`, you have the current status, which should be just fine if you want to do something about it, like powering on.

**Reply by ortegafernando** [reply]
> Why are you even looking so much for that `{"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}"` message? I'd just ignore it, unless you see some very specific good argument for even worrying about that one.
> 
> If you get the `...SENSOR` payload, including `"Power":0`, you have the current status, which should be just fine if you want to do something about it, like powering on.

This ZBMINI always boots up in OFF state. If we want a PowerOnState functionality, we must know when it is booting up. PowerOnState functionality will be very easy if it is "OFF", because it is the default behaviour

This message seems to be the "key" to know that it must boot up recently. {"ZbEZSPReceived":"9B002D69F9FEFFF9E3B406"}"

May be to have a PowerOnState = ON  is impossibe or very difficult.

May be I have to settle for only have a "force ON" functionality by sending every X minutes POWER ON to ZBMINI. But it is a pitty, because I couldn't never have a OFF STATE in ZBMINI with a POWERONSTATE = ON at boot. 

May be it is a dead end.

Thanks a lot for your comments.

**Reply by sfromis** [reply]
Did you decode that message to specifically learn what exactly it means? Or are you simply reading too much into seeing "something"? Maybe it appear in other situations too, meaning that it is not a key to detecting: It just booted. Making ad hoc "seems" assumptions is just too dirty for me, without knowing the background.

What I've already suggested indirectly as a much cleaner setup would be to keep track of your desired state, and then react when the incoming state differs from the desired state. 

More generally, if you need specific control over device operation details, wifi devices where you can flash Tasmota on to the device provides much better options than Zigbee removing detailed control.

**Reply by ortegafernando** [reply]
> Did you decode that message to specifically learn what exactly it means? Or are you simply reading too much into seeing "something"? Maybe it appear in other situations too, meaning that it is not a key to detecting: It just booted. Making ad hoc "seems" assumptions is just too dirty for me, without knowing the background.
> 
> What I've already suggested indirectly as a much cleaner setup would be to keep track of your desired state, and then react when the incoming state differs from the desired state.
> 
> More generally, if you need specific control over device operation details, wifi devices where you can flash Tasmota on to the device provides much better options than Zigbee removing detailed control.

Thanks, I understand you. Keep track of power on/off commands that I send, and if the incoming state differs from the desired state, probably it is because the ZBMINI has restarted.

I think it is the best way.

Thanks  a lot.

**Comment by utech-git** [comment]
Ok, guys. I found out that on [this](https://aliexpress.com/item/1005003026962537.html) switch module I was using (Avatto zigbee) i could change the "power on state" on zigbee2mqtt and on ZHA.

But when I tried the `zbsend {"device":"Switch1","manuf":"0x1002","write":{"StartUpOnOff":0}}` I still got the "UNSUPPORTED_ATTRIBUTE" message.

So I did a little sniffing on the zigbee network while sending the poweronstate with z2m and after a few tries I found out the command to do this on zigbee2tasmota:

`zbsend {"Device": "DEVICE_NAME", "Endpoint": 1, "write":{"0006/8002%30":"0x02"}}`

For what I understood, the 0006 is the cluster (On/Off cluster), the 8002 is the attribute we want to change (the Power On State), I don't know what the "%30" stands for, and the "02" is the "last state" value. Change the 02 to 01 and 00 if you want the Power On State to be on or off, respectively:

`zbsend {"Device": "DEVICE_NAME", "Endpoint": 1, "write":{"0006/8002%30":"0x00"}}` for off
`zbsend {"Device": "DEVICE_NAME", "Endpoint": 1, "write":{"0006/8002%30":"0x01"}}` for on
`zbsend {"Device": "DEVICE_NAME", "Endpoint": 1, "write":{"0006/8002%30":"0x02"}}` for last state

I hope someone that understands how this works could merge this in to the firmware.

**Reply by s-hadinger** [reply]
Attribute 0x8002 is device specific. The good news is that there is now a plugin mechanism to easily define these aliases.

You can add a file named `TS0012_switch.zb` in the file system with the following content (use the Tasmota online editor to create the file - and use a very recent Z2T version from development branch):

```
#Z2Tv1
# TuYa generic switch https://www.zigbee2mqtt.io/devices/TS0012_switch_module.html
:TS0012,_TZ3000_jl7qyupf
:TS0012,_TZ3000_nPGIPl5D
:TS0012,_TZ3000_4zf0crgo
0006/8002%30,StartUpOnOff,manuf:1002
```

With this plugin loaded, you should be able to do simply `zbsend {"device":"Switch1","write":{"StartUpOnOff":0}}` and the command is converted to the right manuf code and attribute for these specific devices only.

**Reply by s-hadinger** [reply]
Please confirm it works, since I can't try it myself. If ok, I can add to the repository

**Reply by utech-git** [reply]
I tried a couple of times, but everytime I flashed the compiled firmware I ended up with my sonoff zb bridge just blinking the blue light. Not connecting to wifi nor prividing the AP for connection.

I think I am doing something wrong when compiling, but I dont know what.

I opened tasmota development branch on github. I created the new file TS0012_switch.zb on the tasmota/zigbee directory.
On plataformio_override.ini I comented the tasmota default_envs and uncommented the tasmota-zbbridge.
Compiled it with the command "platformio run -e tasmota" and download the firmware from directory build_output/firmware/tasmota.bin.

Am I doing anything wrong?

**Reply by sfromis** [reply]
To try with the `.zb` file, you do not need to make a build, or put the file into the tasmota/zigbee directory. Simply put the file in the local file system of the Tasmota device, it will be loaded next restart, as long as your Tasmota firmware version is new enough, like current dev precompiled builds.

If you want to make a build for other reasons (not needed for this), I think that an issue is that when you select a build environment on the platformio run command, it will supersede what you may have done in the override file, meaning that you did not build tasmota-zbbridge, and then used the wrong binary without Zigbee support.

**Reply by s-hadinger** [reply]
Sorry for the confusion, just use the online firmware from development:

ZBBridge: http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz

ZBBridgePro: http://ota.tasmota.com/tasmota32/tasmota32-zbbrdgpro.bin

**Reply by utech-git** [reply]
Oh, OK!!!
Thanks for the clatification. But still no luck.

I updated my zbbridge to developement (version 12.1.1.4), created the files on the filesystem and restarted it. Here's the pictures so you can check if I did anything wrong:

![image](https://user-images.githubusercontent.com/27909006/194715739-e6527e91-43c2-49d2-8250-49c0279ab03d.png)

![image](https://user-images.githubusercontent.com/27909006/194715744-06712a8a-149e-410c-b0a9-41049448c9d3.png)


But I still get the UNSUPPORTED_ATRIBUTES
Command: `zbsend {"device":"0x135D","write":{"StartUpOnOff":0}}`
Output: `12:38:30.374 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0x0006","srcaddr":"0x135D","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0," LinkQuality":92,"securityuse":0,"seqnumber":222,"fc":"0x18","frametype":0,"direction":1,"disableresp":1,"manuf":"0x0000","transact":5, "cmdid":"0x04","payload":"860340"}}`
`12:38:30.379 MQT: tele/zigbeebridge/RESULT = {"ZbResponse":{"Device":"0x135D","Command":"0006_04","Status":134,"StatusMessage":"UNSUPPORTED_ATTRIBUTE","Endpoint":1,"LinkQuality":92}}`

One thing I noticed when I use the other command is that a few "attributes" are different: "fc", "disableresp", "cmdid" and "payload", if this helps you guys debug it in any way:
Command: `zbsend {"Device": "0x135D", "Endpoint": 1, "write":{"0006/8002%30":"0x00"}}`
Output: `12:39:20.348 ZIG: {"ZbZCLReceived":{"groupid":0,"clusterid":"0x0006","srcaddr":"0x135D","srcendpoint":1,"dstendpoint":1,"wasbroadcast":0,"LinkQuality":100,"securityuse":0,"seqnumber":224,"fc":"0x08","frametype":0,"direction":1,"disableresp":0,"manuf":"0x0000","transact":46,"cmdid":"0x0A","payload":"02803000"}}`

Let me know if I did anything wrong or if I could do any other thing to help.

Edit: If this also helps, here's the ZigbeeZCLSend for each command:
Command: `zbsend {"Device": "0x135D", "Endpoint": 1, "write":{"0006/8002%30":"0x00"}}`
`ZigbeeZCLSend device: 0x135D, endpoint:1, cluster:0x0006, cmd:0x0B, send:"0A00"`

Command: `zbsend {"device":"0x135D","write":{"StartUpOnOff":0}}`
`ZigbeeZCLSend device: 0x135D, endpoint:1, cluster:0x0006, cmd:0x02, send:"03403000"`

**Reply by s-hadinger** [reply]
Can you please paste the result from `ZbInfo 0x135D`

Also can you check the logs at the end of Zigbee initialization that the .zb file was correctly loaded ?

**Reply by s-hadinger** [reply]
For the record, I artificially create a virtual device with:

```
backlog ZbForget 0x135D; ZbRestore {"Device":"0x135D","ModelId":"TS0012","Manufacturer":"_TZ3000_jl7qyupf","Endpoints":[1]}
```

Then using `zbsend {"device":"0x135D","write":{"StartUpOnOff":0}}` I got the following log:

```
18:16:02.490 ZigbeeZCLSend device: 0x135D, endpoint:1, cluster:0x0006, cmd:0x02, send:"02803000"
18:16:02.492 ZIG: ZbZNPSent 2402025D1300000000000001000001060001301E0900140210010202803000
```

Which indicates that it is using attribute 8003 as expected.

**Reply by utech-git** [reply]
Sure!
Here's the data:

`{"ZbInfo":{"0x135D":{"Device":"0x135D","IEEEAddr":"0xA4C138AD0C2F5442","ModelId":"TS0001","Manufacturer":"_TZ3000_tqlv4ug4","Endpoints":[1,242],"Config":["O01"],"Power":0,"Reachable":true,"LastSeen":37,"LastSeenEpoch":1665255945,"LinkQuality":123}}} (retained)`

`16:07:58.337 ZIG: ZbLoad 'TS0012_switch.zb' loaded successfully`

With these info I got it working. Here's what I changed:

The manufacturer is `_TZ3000_tqlv4ug4`, ModelID is `TS0001` and manuf is `0000`

So the working setup is:

file named `TS0001_switch.zb` instead of `TS0012_switch.zb`.
(The AliExpress link I sent has all the 1/2/3/4 gang switch. Apparently, TS0012 is the 2 gang and TS0001 is the 1gang. I am testing it with the 1 gang switch).
Whenever I got the UNSUPPORTED_ATTRIBUTE message, the "manuf" was `0000`, so I changed that too

Content of this file is
```
#Z2Tv1
# TuYa generic switch https://www.zigbee2mqtt.io/devices/TS0001_switch_module.html
:TS0001,_TZ3000_tqlv4ug4
0006/8002%30,StartUpOnOff,manuf:0000
```

Here's the output:
`{"ZbReceived":{"0x135D":{"Device":"0x135D","StartUpOnOff":0,"Endpoint":1,"LinkQuality":128}}} (retained)`

Thanks a lot!

**Reply by s-hadinger** [reply]
Great. You can remove the manuf part if it's 0000

**Reply by s-hadinger** [reply]
Thanks this is super useful, I have added a way to have wildcard in model or manufacturer ids. So now the file content is:

```
#Z2Tv1
# TuYa generic switch https://www.zigbee2mqtt.io/devices/TS0001_switch_module.html
:TS0001,_TZ3000_*
0006/8002%30,StartUpOnOff
```

And merged to Tasmota
